libunwind/src/Registers.hpp

```diff
@@ -1873,31 +1885,69 @@ inline bool Registers_arm64::validRegister(int regNum) const {
   return true;
 }
 
-inline uint64_t Registers_arm64::getRegister(int regNum) const {
+inline uintptr_t Registers_arm64::getRegister(int regNum) const {
   if (regNum == UNW_REG_IP)
     return _registers.__pc;
   if (regNum == UNW_REG_SP)
     return _registers.__sp;
   if (regNum == UNW_ARM64_RA_SIGN_STATE)
     return _registers.__ra_sign_state;
+#ifdef __CHERI_PURE_CAPABILITY__
+  if ((regNum >= 0) && (regNum < 32))
+    return (uint64_t)_registers.__x[regNum];
+  if ((regNum >= UNW_ARM64_C0) && (regNum <= UNW_ARM64_C31))
+    return _registers.__x[regNum - UNW_ARM64_C0];
+#else
   if ((regNum >= 0) && (regNum < 32))
     return _registers.__x[regNum];
+#endif
   _LIBUNWIND_ABORT("unsupported arm64 register");
 }
 
-inline void Registers_arm64::setRegister(int regNum, uint64_t value) {
+inline void Registers_arm64::setRegister(int regNum, uintptr_t value) {
   if (regNum == UNW_REG_IP)
     _registers.__pc = value;
   else if (regNum == UNW_REG_SP)
     _registers.__sp = value;
   else if (regNum == UNW_ARM64_RA_SIGN_STATE)
     _registers.__ra_sign_state = value;
+#ifdef __CHERI_PURE_CAPABILITY__
+  if ((regNum >= 0) && (regNum < 32))
+    _registers.__x[regNum] = (uint64_t)value;
+  if ((regNum >= UNW_ARM64_C0) && (regNum <= UNW_ARM64_C31))
+    _registers.__x[regNum - UNW_ARM64_C0] = value;
+#else
   else if ((regNum >= 0) && (regNum < 32))
     _registers.__x[regNum] = value;
+#endif
```

Comment:
```suggestion
  else if ((regNum >= 0) && (regNum < 32))
    _registers.__x[regNum] = (uint64_t)value;
#ifdef __CHERI_PURE_CAPABILITY__
  else if ((regNum >= UNW_ARM64_C0) && (regNum <= UNW_ARM64_C31))
    _registers.__x[regNum - UNW_ARM64_C0] = value;
#endif
```
Ditto (also missing `else`s, otherwise you hit the _LIBUNWIND_ABORT for some valid cases)